src/sqlfluff/rules/L016.py

```diff
@@ -369,6 +369,44 @@ def _gen_line_so_far(raw_stack, initial_buff=None):
                 break  # pragma: no cover
         return working_buff
 
+    @classmethod
+    def _compute_segment_length(cls, segment):
+        if segment.is_type("newline"):
+            # Generally, we won't see newlines, but if we do, simply ignore
+            # them. Rationale: The intent of this rule is to enforce maximum
+            # line length, and newlines don't make lines longer.
+            return 0
+        # Compute the length of this segments in SOURCE space (before template
+        # expansion).
+        slice_length = (
+            segment.pos_marker.source_slice.stop - segment.pos_marker.source_slice.start
+        )
+        if slice_length:
+            return slice_length
+        else:
+            # If a segment did not originate from the original source, its slice
+            # length slice length will be zero. This occurs, for example, when
+            # other lint rules add indentation or other whitespace. In that
+            # case, compute the length of its contents.
+            return len(segment.raw)
+
+    @classmethod
+    def _compute_source_length(cls, segments):
+        line_len = 0
+        seen_slices = set()
+        for segment in segments:
+            slice = (
+                segment.pos_marker.source_slice.start,
+                segment.pos_marker.source_slice.stop,
+            )
+            # Often, a single templated area of a source file will expand to
+            # multiple SQL tokens.Here, we use a set to avoid double counting
+            # the length of that text.
```

Comment:
Is this due to FOR loops for example? Is it worth mentioning that to explain it?

```suggestion
            # Often, a single templated area of a source file will expand to
            # multiple SQL tokens (for example when a For loop expands multiple
            # times). We use a set to avoid double counting the length of that
            # pre-templated text.
```

Or, if not, then you at least need a space between the sentences.

Reply:
I'll add an example here to the comment. 👍

Reply:
That's much clearer now. Wasn't getting that before. Thanks.